File: leetcode/python/0217_contains_duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        nums.sort()
        # 1 1 2 3
        if len(nums) == 1: return False
        # loop through using i (0, len(nums) - 1) and j (1, len(nums) - 1):
        # if nums[i] == nums[j]: return True
        # return False
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]: return True
        
        return False

File: leetcode/cpp/0217_contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> occurred;

        for (int i = 0; i < nums.size(); i++) {
            if (occurred.find(nums[i]) == occurred.end()) {
                occurred.insert(nums[i]);
            }
            else {
                return true;
            }
        }

        return false;
    }
};